go/mysql/datetime/datetime.go

```diff
@@ -521,28 +546,124 @@ func (dt DateTime) Round(p int) (r DateTime) {
 	r = dt
 	if n == 1e9 {
 		r.Time.nanosecond = 0
-		return FromStdTime(r.ToStdTime(time.Local).Add(time.Second))
+		return NewDateTimeFromStd(r.ToStdTime(time.Local).Add(time.Second))
 	}
 	r.Time.nanosecond = uint32(n)
 	return r
 }
 
-func FromStdTime(t time.Time) DateTime {
+func (dt DateTime) toSeconds() int {
+	return (dt.Date.Day()-1)*secondsPerDay + dt.Time.toSeconds()
+}
+
+func (dt *DateTime) addInterval(itv *Interval) bool {
+	switch {
+	case itv.unit.HasTimeParts():
```

Comment:
I'm should assume these conditions are mutually exclusive, right?

Reply:
They're not fully mutually exclusive (an interval with a day can also have hours in it), but they've been arranged in the proper priority order so that the first one that matches implements the addition logic that we're expecting. :ok_hand: